core-services/src/main/java/org/zalando/nakadi/service/publishing/EventPublisher.java

```diff
@@ -124,6 +133,8 @@ EventPublishResult processInternal(final String events,
             if (!delete) {
                 enrich(batch, eventType);
             }
+            eventOwnerExtractor.extract(batch, eventType);
```

Comment:
I see that you do iteration over batch on this and next line. it looks like it could be combined without loosing readability, but reducing cpu cycles on publishing path.

Reply:
@adyach 
The full code is 
```
validate(batch, eventType, parentSpan, delete);
partition(batch, eventType);
setEventKey(batch, eventType);
enrich(batch, eventType);
eventOwnerExtractor.extract(batch, eventType);
authorizeEventWrite(batch);
```
I think there it was done because of readability.

Reply:
I am talking only about your changes, and it is clearly can be improved

Reply:
I thought that I answered why it is done like that. There are 2 things happening here - extraction of header and check of authorization. Logically they are different. 
In case if we are joining extraction with checks, then the method will be called
`extractAndCheckAuthorization`, that didn't seem right logically, but can reduce CPU usage because of failfast approach and iteration removal. 

To be honest, I do not have any preference between this 2 approaches (clear logic vs failfast in this case). 
Anyways, I am not the owner of this PR, and my commit came as a result of pairprogramming

Reply:
Tried to fix this in https://github.com/zalando/nakadi/pull/1147/commits/dfb2c00f98ce6224e624ee6c28445612a6c82ccb